tests/FSharpLint.Core.Tests/Rules/Conventions/CyclomaticComplexity.fs

```diff
@@ -0,0 +1,258 @@
+            let errorLocation = 3, TestConventionsCyclomaticComplexity.Indent.Length
+            yield matchExpression ind nl num, errorLocation
+            yield matchExpressionWithCombinedPatterns ind nl num, errorLocation
+            yield matchFunction ind nl num, errorLocation
+            
+            /// match! expression
+            let errorLocation = 4, TestConventionsCyclomaticComplexity.Indent.Length
+            yield matchBang ind nl num, errorLocation
+            
+            /// boolean conditionals
+            let errorLocation = 3, TestConventionsCyclomaticComplexity.Indent.Length
+            yield ifThenExpressionWithMultipleAndConditionals  ind nl num, errorLocation
+            yield ifThenExpressionWithMultipleOrConditionals ind nl num, errorLocation
+            yield whileWithBooleanOperatorsInConditionExpressions ind nl num, errorLocation    
+        } |> Seq.map (fun (x, y) -> [| box x; box y |])
+ 
+    /// Verifies that no cyclomatic complexity over-maximum flags are raised on source that has cyclomatic complexity <= maxComplexity.
+    [<TestCaseSource(nameof(TestConventionsCyclomaticComplexity.AtMaxComplexityTestCasesSource))>]
+    member this.EnsureNoErrorWhenComplexityBelowThreshold(code) =
+        this.Parse code
+        Assert.IsFalse(this.ErrorsExist)
+      
+    /// Verifies that flags are raised on source code that has cyclomatic complexity > maxComplexity.
+    [<TestCaseSource(nameof(TestConventionsCyclomaticComplexity.FailureCasesSource))>]
+    member this.EnsureErrorWhenComplexityExceedsThreshold(code, errorLocation) =
+        this.Parse code
+        Assert.IsTrue(this.ErrorExistsAt(errorLocation))
+    
+    /// Verifies that an error is raised on a match expression which has multiple boolean operator conditions in multiple when clauses.
+    [<Test>]
+    member this.TestLogicalOperatorsInMatch() =
+        let indent = TestConventionsCyclomaticComplexity.Indent
+        let newline = TestConventionsCyclomaticComplexity.NewLine
+        let code = "Module Program" + newline + 
+                   "let f() = " + newline +
+                   makeMatchSnippetWithLogicalOperatorsInWhenClause indent newline (maxComplexity + 1) 
+        this.Parse code
+        Assert.AreEqual(1, this.ErrorRanges.Length) 
+    
+    /// Verifies that an error is raised on a single match clause with multiple logical operators.
+    [<Test>]
+    member this.TestMultipleLogicalOperatorsInSingleMatchClause() =
+        let indent = TestConventionsCyclomaticComplexity.Indent
+        let newline = TestConventionsCyclomaticComplexity.NewLine
+        let code = "Module Program" + newline + 
+                   "let f() = " + newline +
+                   indent + "match \"dummyString\" with" + newline +
+                   indent + "| x when x = \"1\"" + ([2..maxComplexity+1] |> List.map (sprintf "|| x = \"%d\"") |> String.concat " ") + " -> ()" + newline +
+                   indent + "| _ -> ()"
```

Comment:
question, why do you use things like `newline` and `indent` here instead of having a multi-line string where you place them yourself? (the latter would be much more readable, and you can still interpolate other things with %s AFAIU)

Reply:
Yes, okay, agreed. I will make that change.

Are there any other changes you'd like?

Reply:
> Yes, okay, agreed. I will make that change.

Cool.

> Are there any other changes you'd like?

Not from my side. I'm not the maintainer, just started contributing recently.